src/main/java/soot/jimple/validation/ReturnStatementsValidator.java

```diff
@@ -25,63 +25,66 @@
 import java.util.List;
 
 import soot.Body;
+import soot.SootMethod;
 import soot.Unit;
 import soot.VoidType;
+import soot.baf.GotoInst;
+import soot.baf.ReturnInst;
+import soot.baf.ReturnVoidInst;
+import soot.baf.ThrowInst;
 import soot.jimple.GotoStmt;
-import soot.jimple.RetStmt;
 import soot.jimple.ReturnStmt;
 import soot.jimple.ReturnVoidStmt;
 import soot.jimple.ThrowStmt;
 import soot.validation.BodyValidator;
 import soot.validation.ValidationException;
 
+/**
+ * Checks that this Body actually contains a throw or return statement, and that the return statement is of the appropriate
+ * type (i.e. void/non-void).
+ */
 public enum ReturnStatementsValidator implements BodyValidator {
   INSTANCE;
 
   public static ReturnStatementsValidator v() {
     return INSTANCE;
   }
 
-  /**
-   * Checks the following invariants on this Jimple body:
-   * <ol>
-   * <li>this-references may only occur in instance methods
-   * <li>this-references may only occur as the first statement in a method, if they occur at all
-   * <li>param-references must precede all statements that are not themselves param-references or this-references, if they
-   * occur at all
-   * </ol>
-   */
   @Override
   public void validate(Body body, List<ValidationException> exceptions) {
-    // Checks that this Jimple body actually contains a return statement, and that the return statement
-    // is of the appropriate type (i.e. void/non-void)
+    final SootMethod method = body.getMethod();
+
+    // Checks that this Body actually contains a throw or return statement, and
+    // that the return statement is of the appropriate type (i.e. void/non-void)
     for (Unit u : body.getUnits()) {
-      if ((u instanceof RetStmt) || (u instanceof ThrowStmt)) {
```

Comment:
```RetStmt``` is not a method return, it is the deprecated ```ret``` opcode (that pairs with jsr).

Reply:
yeah, I agree. Why is this checked here